drivers/i2c/i2c_gpio.c

```diff
@@ -142,7 +142,7 @@ static int i2c_gpio_init(struct device *dev)
 	}
 
 	err = gpio_config(context->sda_gpio, config->sda_pin,
-			  config->sda_flags | GPIO_OUTPUT_HIGH);
+			  config->sda_flags | GPIO_INPUT | GPIO_OUTPUT_HIGH);
```

Comment:
I suspect this will render the I2C GPIO bit banging driver useless on most GPIO controllers, since they fail configuration if both GPIO_INPUT and GPIO_OUTPUT are specified.

I tested the existing implementation on the FRDM-K64F using `PORT_PCR_ODE_MASK` pinmux configuration, which puts the given pin in open-drain mode. It's configured as an output, but the open drain configuration allows it to be pulled down - and that real value is read back.

One could argue, that the GPIO controller driver should allow GPIO_INPUT | GPIO_OUTPUT when the pin is configured as open-drain, but the GPIO controller driver has no easy way of knowing the pinmux configuration.

Reply:
I've updated this so if configuring bidirectional fails it should fall back to the old way.